test/distributed/_tensor/test_random_ops.py

```diff
@@ -0,0 +1,131 @@
+    def test_device_mesh_init(self):
+        # device mesh init should sync seed and store it as an attribute
+        # However, we have not figured out how we want to use the seed, so
+        # temporarily device_mesh._seed will just equal to each rank's
+        # `initial_seed()`.
+        torch.cuda.manual_seed(self.rank)
+        object_list = [torch.cuda.initial_seed()]
+        broadcast_object_list(object_list)
+        seed_from_rank_0 = int(object_list[0])
+
+        device_mesh = DeviceMesh(self.device_type, torch.arange(self.world_size))
+        # TODO: sync seed once we figure out how we want to use the seed
+        if self.rank != 0:
+            with self.assertRaises(AssertionError):
+                self.assertEqual(seed_from_rank_0, device_mesh._seed)
+
+    @with_comms
+    @skip_unless_torch_gpu
+    def test_manual_seed(self):
+        device_mesh = DeviceMesh(self.device_type, torch.arange(self.world_size))
+        manual_seed(1234, device_mesh)
+        with self.assertRaisesRegex(RuntimeError, "different seed values"):
+            manual_seed(self.rank, device_mesh)
+
+    @with_comms
+    @skip_unless_torch_gpu
+    def test_deterministic_uniform_1d(self):
+        def check_rng_state(seed: int, offset: int, device_mesh: DeviceMesh) -> None:
+            state = get_rng_state(device_mesh)
+            seed_int64 = state[-16:-8].view(torch.int64)
+            offset_int64 = state[-8:].view(torch.int64)
+            self.assertEqual(seed_int64, torch.tensor([seed]))
+            self.assertEqual(offset_int64, torch.tensor([offset]))
+
+        device_mesh = DeviceMesh(self.device_type, torch.arange(self.world_size))
+        size = [4, 1]
+
+        # initialize rng state
+        manual_seed(1234, device_mesh)
+        check_rng_state(1234, 0, device_mesh)
+        _tensor = torch.empty(*size, device="cuda")
+        dtensor = DTensor.from_local(_tensor, device_mesh, [Shard(1)])
+
+        # preprocess rng offset
+        global_size = dtensor.numel()
+        local_size = dtensor.to_local().numel()
+        state = get_rng_state(device_mesh)
+        offset = state[-8:].view(torch.int64)[0].item()
+        _set_offset(offset + self.rank * local_size, device_mesh)
+        check_rng_state(1234, self.rank * local_size, device_mesh)
+
+        # random op call
+        dtensor.uniform_(0, 1)
+
+        # postprocess rng offset
+        _set_offset(offset + global_size, device_mesh)
+        check_rng_state(1234, global_size, device_mesh)
+
+        # allgather the local tensors
+        dtensor = dtensor.redistribute(device_mesh, [Replicate()])
+        local_tensor = dtensor.to_local()
+
+        # compare with local tensors from other ranks
+        for other_rank in range(self.world_size):
+            if self.rank != other_rank:
+                # other rank should have a different local tensor
+                self.assertNotEqual(
+                    local_tensor[:, self.rank], local_tensor[:, other_rank]
```

Comment:
hmmm is this really correct? if self.rank != other_rank, `local_tensor[:, self.rank]` and `local_tensor[:, other_rank]` would have different shape, so they are definitely not equal. I guess you are not testing what you want, you probably want to do things like `local_tensor_list = local_tensor.tensor_split(4)`, then compare the list of tensors instead.

Reply:
Let me quickly double check it

Reply:
resolved offline